policies/examples/time_based.rego

```diff
@@ -1,12 +1,31 @@
 package examples.time_based
 
+# This policy always fails
+# METADATA
+# custom:
+#   effective_on: 2099-05-02T00:00:00Z
+
 # When effective_on is specified, the policy will be ignored until the day/time is reached.
 # Use "effective_on := time.now_ns()", or omit the variable declaration, to make the policy
 # always applicable.
-effective_on := time.parse_rfc3339_ns("2099-05-02T00:00:00Z")
+deny[{"msg": msg, "effective_on": rego.metadata.rule().custom.effective_on}] {
+	true
+	msg := "Roads?"
+}
 
-# This policy always fails
+# This policy always fails, but doesn't have effective_on set
+# METADATA
```

Comment:
Oh it's interesting that the metadata can be applied to a specific rule, and not the whole file (module?).

Reply:
Annotations can be defined on different [scopes](https://www.openpolicyagent.org/docs/v0.40.0/annotations/#scope). Added a bit to explain this in ec25eef6a094042bb56ef1b2fdbd67f3a9c63712, all based on [Metadata Merge strategies](https://www.openpolicyagent.org/docs/v0.40.0/policy-reference/#metadata-merge-strategies).